backend/service/authn/oidc.go

```diff
@@ -0,0 +1,241 @@
+package authn
```

Comment:
Is this all moved code? Or is there anything you would like specifically reviewed?

Reply:
yes, with the exception of a couple minor argument changes it's a refactor + new test code